libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/StreamingExtensions/BotFrameworkStreamingExtensionsAdapter.cs

```diff
@@ -0,0 +1,321 @@
+        {
+            if (activity == null)
+            {
+                throw new ArgumentNullException(nameof(activity));
+            }
+
+            var route = string.Format("/v3/conversations/{0}/activities/{1}", conversationId, activity.Id);
+            var request = Request.CreatePost(route);
+            request.SetBody(activity);
+
+            return await SendRequestAsync<ResourceResponse>(request).ConfigureAwait(false);
+        }
+
+        public async Task<HttpOperationResponse> DeleteActivityAsync(string conversationId, string activityId, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var route = string.Format("/v3/conversations/{0}/activities/{1}", conversationId, activityId);
+            var request = Request.CreateDelete(route);
+
+            return await SendRequestAsync<HttpOperationResponse>(request).ConfigureAwait(false);
+        }
+
+        public async Task<IList<ChannelAccount>> GetConversationMembersAsync(string conversationId, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var route = string.Format("/v3/conversations/{0}/members", conversationId);
+            var request = Request.CreateGet(route);
+
+            return await SendRequestAsync<IList<ChannelAccount>>(request).ConfigureAwait(false);
+        }
+
+        public async Task<PagedMembersResult> GetConversationPagedMembersAsync(string conversationId, int? pageSize = null, string continuationToken = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var route = string.Format("/v3/conversations/{0}/pagedmembers", conversationId);
+            var request = Request.CreateGet(route);
+
+            return await SendRequestAsync<PagedMembersResult>(request).ConfigureAwait(false);
+        }
+
+        public async Task<HttpOperationResponse> DeleteConversationMemberAsync(string conversationId, string memberId, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var route = string.Format("/v3/conversations/{0}/members/{1}", conversationId, memberId);
+            var request = Request.CreateDelete(route);
+
+            return await SendRequestAsync<HttpOperationResponse>(request).ConfigureAwait(false);
+        }
+
+        public async Task<IList<ChannelAccount>> GetActivityMembersAsync(string conversationId, string activityId, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var route = string.Format("/v3/conversations/{0}/activities/{1}/members", conversationId, activityId);
+            var request = Request.CreateGet(route);
+
+            return await SendRequestAsync<IList<ChannelAccount>>(request).ConfigureAwait(false);
+        }
+
+        private async Task<T> SendRequestAsync<T>(Request request)
+        {
+            try
+            {
+                var serverResponse = await _server.SendAsync(request).ConfigureAwait(false);
```

Comment:
I'm adding a cancellationToken parameter here...you'll want to pass one in, and then catch "TaskCancelledException" and deal with timeouts